pkg/csi/service/common/commonco/k8sorchestrator/topology.go

```diff
@@ -1517,14 +1532,39 @@ func (volTopology *wcpControllerVolumeTopology) GetSharedDatastoresInTopology(ct
 		}
 
 		// Call GetCandidateDatastores for each cluster moref. Ignore the vsanDirectDatastores for now.
-		for _, clusterMoref := range clusterMorefs {
-			accessibleDs, _, err := cnsvsphere.GetCandidateDatastoresInCluster(ctx, params.Vc, clusterMoref)
+		if !isPodVMOnStretchedSupervisorEnabled {
+			accessibleDs, _, err := cnsvsphere.GetCandidateDatastoresInCluster(ctx, params.Vc, clusterMorefs[0])
 			if err != nil {
 				return nil, logger.LogNewErrorf(log,
 					"failed to find candidate datastores to place volume in cluster %q. Error: %v",
-					clusterMoref, err)
+					clusterMorefs[0], err)
 			}
 			sharedDatastores = append(sharedDatastores, accessibleDs...)
+		} else {
+			var sharedDatastoresForclusterMorefs []*cnsvsphere.DatastoreInfo
```

Comment:
can we add comment for readability here to indicate that this logic finds datastore shared across all clusters within the zone?

Reply:
done.